syntax = "proto3";


package freo.me.purchase;

service Purchase {
 
  rpc purchase (PurchaseRequest) returns (PurchaseReply) {}
  rpc push (Empty) returns (stream Update) {}
}

message Empty {}

message Update {
  string msg = 1;
}

message PurchaseRequest {
  string poNumber = 1;
  string lineItem = 2;
  int32 quantity = 3;
  Date date = 4;
  string customerNumber= 5;
  string paymentReference = 6;

}

message Date {
	int32 year = 1;
	int32 month = 2;
	int32 day = 3;
}

message PurchaseReply {
  string uuid = 1;
  int32 returncode = 2;
}
